gemini/use-cases/retail/product_attributes_extraction.ipynb

```diff
@@ -0,0 +1,895 @@
+      },
+      "source": [
+        "### Authenticate your notebook environment (Colab only)\n",
+        "\n",
+        "If you're running this notebook on Google Colab, run the cell below to authenticate your environment."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 1,
+      "metadata": {
+        "id": "NyKGtVQjgx13"
+      },
+      "outputs": [],
+      "source": [
+        "import sys\n",
+        "\n",
+        "if \"google.colab\" in sys.modules:\n",
+        "    from google.colab import auth\n",
+        "\n",
+        "    auth.authenticate_user()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "DF4l8DTdWgPY"
+      },
+      "source": [
+        "### Set Google Cloud project information and initialize Vertex AI SDK\n",
+        "\n",
+        "To get started using Vertex AI, you must have an existing Google Cloud project and [enable the Vertex AI API](https://console.cloud.google.com/flows/enableapi?apiid=aiplatform.googleapis.com).\n",
+        "\n",
+        "Learn more about [setting up a project and a development environment](https://cloud.google.com/vertex-ai/docs/start/cloud-environment)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 2,
+      "metadata": {
+        "id": "Nqwi-5ufWp_B"
+      },
+      "outputs": [],
+      "source": [
+        "PROJECT_ID = \"cloud-llm-preview1\"  # @param {type:\"string\"}\n",
+        "LOCATION = \"us-central1\"  # @param {type:\"string\"}\n",
+        "\n",
+        "\n",
+        "import vertexai\n",
+        "\n",
+        "vertexai.init(project=PROJECT_ID, location=LOCATION)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "EdvJRUWRNGHE"
+      },
+      "source": [
+        "## Build a Product Image Agent"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 11,
+      "metadata": {
+        "cellView": "form",
+        "id": "xJQczGZX3FfW"
+      },
+      "outputs": [],
+      "source": [
+        "# @title Image loading and parsing library\n",
+        "import json\n",
+        "\n",
+        "import requests\n",
+        "from google.cloud import storage\n",
+        "\n",
+        "\n",
+        "def load_bytes_from_gcs(gcs_url: str) -> bytes:\n",
+        "    \"\"\"Loads the (image) bytes from gcs url.\n",
+        "\n",
+        "    Args:\n",
+        "        gcs_url (str): The url for the google cloud storage object (starts with gs://).\n",
+        "\n",
+        "    Returns:\n",
+        "        The object/image in bytes.\n",
+        "    \"\"\"\n",
+        "    storage_client = storage.Client()\n",
+        "    bucket_name, blob_name = gcs_url.replace(\"gs://\", \"\").split(\"/\", 1)\n",
+        "    bucket = storage_client.bucket(bucket_name)\n",
+        "    blob = bucket.blob(blob_name)\n",
+        "    image_bytes = blob.download_as_bytes()\n",
+        "    return image_bytes\n",
```

Comment:
I have some debug flag that can be used to show the image, prompt and response, so maybe still need this to load the bytes from gcs to display the image in the UI?

Reply:
You can convert the `gs://` uri to `https://storage.googleapis.com/` which can be used as a display link to avoid downloading all of the images. It's generally preferred to use remote images when possible to save on latency.

Reply:
Thanks, do you have an example of displaying an image in colab from url? I tried "IPython.display.Image(url)" but it failed when the url does not have the proper file extension at the end.

Reply:
Actually found a way to use HTML tags instead of image_data. I removed all the load_xxx_from_uri() functions.